Test fail on arches ppc64le and s390x

Hello,

I am in the process of packaging ewma for Fedora but I encountered problems running the tests. On both ppc64le and s390x arches, the following test fails:

```
+ export GOPATH=/builddir/build/BUILDROOT/golang-github-VividCortex-ewma-0-0.1.git4cc8cc5.fc27.ppc64le//usr/share/gocode:/usr/share/gocode
+ GOPATH=/builddir/build/BUILDROOT/golang-github-VividCortex-ewma-0-0.1.git4cc8cc5.fc27.ppc64le//usr/share/gocode:/usr/share/gocode
+ go test -compiler gc -ldflags '' github.com/VividCortex/ewma
--- FAIL: TestSimpleEWMA (0.00s)
	ewma_test.go:26: e.Value() is 4734.500946466117, wanted 4734.500946466118
--- FAIL: TestVariableEWMA (0.00s)
	ewma_test.go:40: e.Value() is 4734.500946466117, wanted 4734.500946466118
FAIL
FAIL	github.com/VividCortex/ewma	0.018s
```

Could you identify why there is a discrepancy in the value given versus the value expected, only on these two arches while others arches test are fine?

--- a/ewma_test.go
+++ b/ewma_test.go
@@ -3,7 +3,12 @@ package ewma
 // Copyright (c) 2013 VividCortex, Inc. All rights reserved.
 // Please see the LICENSE file for applicable license terms.
 
-import "testing"
+import (
+	"math"
+	"testing"
+)
+
+const testMargin = 0.00000001
 
 var samples = [100]float64{
 	4599, 5711, 4746, 4621, 5037, 4218, 4925, 4281, 5207, 5203, 5594, 5149,
@@ -17,12 +22,16 @@ var samples = [100]float64{
 	3197, 5139, 6101, 5279,
 }
 
+func withinMargin(a, b float64) bool {
+	return math.Abs(a-b) <= testMargin
+}
+
 func TestSimpleEWMA(t *testing.T) {
 	var e SimpleEWMA
 	for _, f := range samples {
 		e.Add(f)
 	}
-	if e.Value() != 4734.500946466118 {
+	if !withinMargin(e.Value(), 4734.500946466118) {
 		t.Errorf("e.Value() is %v, wanted %v", e.Value(), 4734.500946466118)
 	}
 	e.Set(1.0)
@@ -36,7 +45,7 @@ func TestVariableEWMA(t *testing.T) {
 	for _, f := range samples {
 		e.Add(f)
 	}
-	if e.Value() != 4734.500946466118 {
+	if !withinMargin(e.Value(), 4734.500946466118) {
 		t.Errorf("e.Value() is %v, wanted %v", e.Value(), 4734.500946466118)
 	}
 	e.Set(1.0)
@@ -50,7 +59,7 @@ func TestVariableEWMA2(t *testing.T) {
 	for _, f := range samples {
 		e.Add(f)
 	}
-	if e.Value() != 5015.397367486725 {
+	if !withinMargin(e.Value(), 5015.397367486725) {
 		t.Errorf("e.Value() is %v, wanted %v", e.Value(), 5015.397367486725)
 	}
 }

